### PR TITLE
fix: message list shouldn't get stuck loading

### PR DESCRIPTION
**Pull request by elisealix22**
## What does this PR do?
This PR makes it so our `useGetConversation` hook listens for updates to `conversations` since it sets the selectedConversation based on the data in that store. This should fix sometimes there being a null selectedConversation which appeared like an endless loading state.

It also polishes the missing xmtp auth UI for people who haven't enabled DMs yet.

| Before | After |
|--|--|
| CleanShot 2022-10-28 at 14 05 25@2x |  Screen Shot 2022-10-28 at 4 03 33 PM |

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- [x] Open a profile of someone who has messages enabled but you haven't DM'd with yet from their profile. Ensure the profile loads.
- [x] Navigate to a profile who hasn't authenticated w XMTP and ensure you can't type a message yet

### CODE REVIEW

File: data/feature-flags.ts [hunk 1]
```diff
@@ -27,7 +27,8 @@ export const featureFlags = [
       '0x016b25', // adamalix.lens
       '0x016b24', // bhavyam.lens
       '0x016b0b', // xmtplabs.lens
-      '0xf5ff' // anoopr.lens
+      '0xf5ff', // anoopr.lens
+      '0x016b2e' // kenziestokes.lens
```
Comment (elisealix22) [inline data/feature-flags.ts:31]: This is another XMTP employee btw!

File: src/components/Messages/Message.tsx [hunk 2]
```diff
@@ -77,7 +77,7 @@ const Message: FC<MessageProps> = ({ conversationKey }) => {
                 hasMore={hasMore}
                 missingXmtpAuth={missingXmtpAuth ?? false}
               />
-              <Composer sendMessage={sendMessage} />
+              <Composer sendMessage={sendMessage} missingXmtpAuth={missingXmtpAuth ?? false} />
```
Comment (neekolas) [inline src/components/Messages/Message.tsx:80]: I would name this something more generic like "disabled", so that the component is more reusable

**Review by neekolas — APPROVED** [review]

File: src/components/Messages/Message.tsx (hunk 2 above)
Comment (elisealix22) [reply src/components/Messages/Message.tsx]: Dig it! Updated.

File: src/components/utils/hooks/useGetConversation.tsx [hunk 3]
```diff
@@ -44,7 +44,7 @@ const useGetConversation = (conversationKey: string, profile?: Profile) => {
     };
     createNewConversation();
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [profile]);
+  }, [profile, conversations]);
```
Comment (neekolas) [inline src/components/utils/hooks/useGetConversation.tsx:47]: This is going to re-run any time any conversation changes. Should probably be scoped to the single conversation we care about.

**Review by bigint — APPROVED** [review]

**Comment by bigint** [comment]
@elisealix22 can you please resolve the conflicts 🙏🏼

File: src/components/utils/hooks/useGetConversation.tsx (hunk 3 above)
Comment (elisealix22) [reply src/components/utils/hooks/useGetConversation.tsx]: Glad you called this out! Updated to only listen for the `selectedConversation`!

**Comment by elisealix22** [comment]
Resolved @bigint ! Sorry for the delay

**Review by neekolas — APPROVED** [review]

**Pull request merged**